src/hotspot/share/memory/universe.cpp

```diff
@@ -580,6 +589,21 @@ oop Universe::out_of_memory_error_realloc_objects() {
   return gen_out_of_memory_error(out_of_memory_errors()->obj_at(_oom_realloc_objects));
 }
 
+bool Universe::is_out_of_memory_error(oop obj, const char* msg) {
+  assert(msg != NULL, "sanity check");
+  oop msg_oop = java_lang_Throwable::message(obj);
+  const char * msg_str = java_lang_String::as_utf8_string(msg_oop);
+  return strcmp(msg, msg_str) == 0;
+}
+
+bool Universe::is_out_of_memory_error_metaspace(oop obj) {
+  return is_out_of_memory_error(obj, MSG_METASPACE);
+}
+
+bool Universe::is_out_of_memory_error_class_metaspace(oop obj) {
+  return is_out_of_memory_error(obj, MSG_CLASS_METASPACE);
+}
+
```

Comment:
You only need the const strings for these two strings, so could you just change these two?